grafana-plugin/src/containers/IntegrationForm/IntegrationForm.tsx

```diff
@@ -32,481 +32,613 @@ import { IntegrationHelper } from 'pages/integration/Integration.helper';
+              <Text type="primary">ServiceNow configuration</Text>
+            </div>
+          </RenderConditionally>
+
+          <RenderConditionally shouldRender={isServiceNow}>
+            <Controller
+              name={FormFieldKeys.ServiceNowUrl}
+              control={control}
+              rules={{ required: 'Instance URL is required', validate: validateURL }}
+              render={({ field }) => (
+                <Field
+                  key={'InstanceURL'}
+                  label={'Instance URL'}
+                  invalid={!!errors[FormFieldKeys.ServiceNowUrl]}
+                  error={errors[FormFieldKeys.ServiceNowUrl]?.message as string}
+                >
+                  <Input {...field} />
+                </Field>
+              )}
+            />
+          </RenderConditionally>
+
+          <RenderConditionally shouldRender={isServiceNow}>
+            <Controller
+              name={FormFieldKeys.AuthUsername}
+              control={control}
+              rules={{ required: 'Username is required' }}
+              render={({ field }) => (
+                <Field
+                  key={'AuthUsername'}
+                  label={'Username'}
+                  invalid={!!errors[FormFieldKeys.AuthUsername]}
+                  error={errors[FormFieldKeys.AuthUsername]?.message as string}
+                >
+                  <Input {...field} />
+                </Field>
+              )}
+            />
+          </RenderConditionally>
+
+          <RenderConditionally shouldRender={isServiceNow}>
+            <Controller
+              name={FormFieldKeys.AuthPassword}
+              control={control}
+              rules={{ required: 'Password is required' }}
+              render={({ field }) => (
+                <Field
+                  key={'AuthPassword'}
+                  label={'Password'}
+                  invalid={!!errors[FormFieldKeys.AuthPassword]}
+                  error={errors[FormFieldKeys.AuthPassword]?.message as string}
+                >
+                  <Input {...field} type="password" />
+                </Field>
+              )}
+            />
+          </RenderConditionally>
+
+          <RenderConditionally shouldRender={isServiceNow}>
+            <Button className={styles.webhookTest} variant="secondary" onClick={onWebhookTestClick}>
+              Test
+            </Button>
+          </RenderConditionally>
+
+          <RenderConditionally shouldRender={isServiceNow}>
+            <Controller
+              name={FormFieldKeys.DefaultWebhooks}
+              control={control}
+              render={({ field }) => (
+                <div className={styles.webhookSwitch}>
+                  <Switch value={field.value} onChange={field.onChange} />
+                  <Text type="primary"> Create default outgoing webhook events</Text>
                 </div>
               )}
+            />
+          </RenderConditionally>
```

Comment:
So we have bunch of things next to each other that should be conditionally rendered for SNOW only. Instead of wrapping each item with `RenderConditionally`, maybe we can wrap them all with it?

Reply:
No I need them child as of `<VerticalGroup>` tree

Reply:
`VerticalGroup` creates a wrapper div for each child it finds, if I render one big chunk of `RenderConditionally`, then that's just 1 child and the content in it is no longer bound to `VerticalGroup` styles basically. I can double check that but afaik that's how it behaves

Reply:
I don't  see a `VerticalGroup` wrapper here but anyway I still think it will work the same. Could you check 
```
          <RenderConditionally shouldRender={isServiceNow}>
            <>
              <div className={styles.serviceNowHeading}>
                <Text type="primary">ServiceNow configuration</Text>
              </div>
              <Controller
                name={FormFieldKeys.ServiceNowUrl}
                control={control}
                rules={{ required: 'Instance URL is required', validate: validateURL }}
                render={({ field }) => (
                  <Field
                    key={'InstanceURL'}
                    label={'Instance URL'}
                    invalid={!!errors[FormFieldKeys.ServiceNowUrl]}
                    error={errors[FormFieldKeys.ServiceNowUrl]?.message as string}
                  >
                    <Input {...field} />
                  </Field>
                )}
              />
              <Controller
                name={FormFieldKeys.AuthUsername}
                control={control}
                rules={{ required: 'Username is required' }}
                render={({ field }) => (
                  <Field
                    key={'AuthUsername'}
                    label={'Username'}
                    invalid={!!errors[FormFieldKeys.AuthUsername]}
                    error={errors[FormFieldKeys.AuthUsername]?.message as string}
                  >
                    <Input {...field} />
                  </Field>
                )}
              />
              <Controller
                name={FormFieldKeys.AuthPassword}
                control={control}
                rules={{ required: 'Password is required' }}
                render={({ field }) => (
                  <Field
                    key={'AuthPassword'}
                    label={'Password'}
                    invalid={!!errors[FormFieldKeys.AuthPassword]}
                    error={errors[FormFieldKeys.AuthPassword]?.message as string}
                  >
                    <Input {...field} type="password" />
                  </Field>
                )}
              />
              <Button className={styles.webhookTest} variant="secondary" onClick={onWebhookTestClick}>
                Test
              </Button>
              <Controller
                name={FormFieldKeys.DefaultWebhooks}
                control={control}
                render={({ field }) => (
                  <div className={styles.webhookSwitch}>
                    <Switch value={field.value} onChange={field.onChange} />
                    <Text type="primary"> Create default outgoing webhook events</Text>
                  </div>
                )}
              />
            </>
          </RenderConditionally>
```

Reply:
you're right, had the impression it was below <VerticalGroup>, I put it all under the conditional check